test/extended/openstack/cpms.go

```diff
@@ -0,0 +1,129 @@
+		o.Expect(err).NotTo(o.HaveOccurred(), "error fetching Masters machines")
+
+		cpmsFlavor := controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.flavor").String()
+		cpmsImage := controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.image").String()
+		cpmsSecurityGroups := make(map[string]struct{})
+		cpmsServerGroup := controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.serverGroupName").String()
+		cpmsSubnets := make(map[string]struct{})
+
+		for _, sg := range objects(controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.securityGroups")) {
+			securityGroupName := sg["name"].(string)
+			cpmsSecurityGroups[securityGroupName] = struct{}{}
+		}
+
+		for _, subnet := range objects(controlPlaneMachineSets[0].Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.networks[0].subnets")) {
+			subnetName := subnet.Get("filter.name").Str()
+			cpmsSubnets[subnetName] = struct{}{}
+		}
+
+		for _, machine := range masterMachines {
+			machineName := machine.Get("metadata.name").String()
+			g.By("Comparing the MachineSet spec with machine " + machineName)
+
+			machineFlavor := machine.Get("spec.providerSpec.value.flavor").String()
+			machineImage := machine.Get("spec.providerSpec.value.image").String()
+			machineSecurityGroups := make(map[string]struct{})
+			machineSubnets := make(map[string]struct{})
+			machineServerGroup := machine.Get("spec.providerSpec.value.serverGroupName").String()
+
+			for _, sg := range objects(machine.Get("spec.providerSpec.value.securityGroups")) {
+				securityGroupName := sg["name"].(string)
+				machineSecurityGroups[securityGroupName] = struct{}{}
+			}
+
+			for _, subnet := range objects(machine.Get("spec.providerSpec.value.networks[0].subnets")) {
+				subnetName := subnet.Get("filter.name").Str()
+				machineSubnets[subnetName] = struct{}{}
+			}
+			o.Expect(machineFlavor).To(o.Equal(cpmsFlavor), "flavor mismatch on Machine %q", machineName)
+			o.Expect(machineImage).To(o.Equal(cpmsImage), "image mismatch on Machine %q", machineName)
+			o.Expect(machineSecurityGroups).To(o.Equal(cpmsSecurityGroups), "security group mismatch on Machine %q", machineName)
+			o.Expect(machineSubnets).To(o.Equal(cpmsSubnets), "subnets mismatch on Machine %q", machineName)
+			o.Expect(machineServerGroup).To(o.Equal(cpmsServerGroup), "server group	 mismatch on Machine %q", machineName)
+		}
+	})
+})
+
+// getControlPlaneMachineSets returns all the available MachineSets
+func getControlPlaneMachineSets(ctx context.Context, dc dynamic.Interface) ([]objx.Map, error) {
+	mc := dc.Resource(schema.GroupVersionResource{
+		Group:    machineAPIGroup,
+		Version:  "v1",
+		Resource: "controlplanemachinesets",
+	}).Namespace(machineAPINamespace)
+	obj, err := mc.List(ctx, metav1.ListOptions{})
```

Comment:
What about we directly `Get` it by name, and return that? Unless we want to test that there's no more than one (and I don't think we want to test that here).

Reply:
Again, I'm confused by this function. It returns a slice, but in the test you directly access `[0]`. If think you should EITHER:
A. `Get` by name instead of `List` here, and return a single CPMS; OR
B. keep returning a slice, but checking in the test that the list has exactly one item